src/providers/common.ts

```diff
@@ -238,3 +239,20 @@ export interface IRunnerProvider extends ec2.IConnectable, iam.IGrantable {
    */
   getStepFunctionTask(parameters: RunnerRuntimeParameters): stepfunctions.IChainable;
 }
+
+/**
+ * Base class for all providers with common methods used by all providers.
+ *
+ * @internal
+ */
+export abstract class BaseProvider extends Construct {
+  protected labelsFromProperties(defaultLabel: string, propsLabel?: string | string[]): string[] {
+    if (typeof propsLabel === 'string') {
+      return [propsLabel];
+    } else if (typeof propsLabel === 'object') {
```

Comment:
will this ever become an object ?
propsLabel can be string | string[]

Reply:
In JavaScript:
```js
> typeof []
'object'